File: app/src/main/java/com/example/moodresetwheel/MainActivity.java
package com.example.moodresetwheel;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        //  Always clear username on fresh app launch
        SharedPreferences prefs = getSharedPreferences("app_prefs", MODE_PRIVATE);
        prefs.edit().remove("username").apply();

        // Always go to Welcome
        startActivity(new Intent(this, BreathingActivity.class));
        finish();
    }
}

File: MoodResetWheel-main copy/app/src/main/java/com/example/moodresetwheel/MainActivity.kt
package com.example.moodresetwheel

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val prefs = getSharedPreferences("app_prefs", MODE_PRIVATE)
        val savedName = prefs.getString("username", null)

        val next = if (savedName.isNullOrBlank()) {
            Intent(this, WelcomeActivity::class.java)
        } else {
            Intent(this, FeelingLogActivity::class.java)
        }

        startActivity(next)
        finish()
    }
}
